src/tools/pkcs11-tool.c

```diff
@@ -7651,17 +7650,13 @@ static int encrypt_decrypt(CK_SESSION_HANDLE session,
 			printf("set md failed, returning\n");
 			return 0;
 		}
-		switch (mgf) {
+		switch (oaep_params.mgf) {
 		case CKG_MGF1_SHA1:
 			md = EVP_sha1();
 			break;
 		case CKG_MGF1_SHA224:
 			md = EVP_sha224();
 			break;
-		default:
-			printf("mgf %s unknown, defaulting to CKG_MGF1_SHA256\n", p11_mgf_to_name(mgf));
-			mgf = CKG_MGF1_SHA256;
-			/* fall through */
```

Comment:
Shouldn't the `default` case for `CKG_MGF1_SHA256` be just moved at the end with setting `md = EVP_sha256()`?

Reply:
The `oaep_params.mgf` is always set by `build_rsa_oaep_params` so I think there is no need to set the default here.